cpp/src/strings/strings_column_factories.cu

```diff
@@ -17,35 +17,165 @@
 #include <cudf/column/column.hpp>
 #include <cudf/column/column_factories.hpp>
 #include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/detail/utilities/vector_factories.hpp>
+#include <cudf/detail/valid_if.cuh>
 #include <cudf/strings/detail/strings_column_factories.cuh>
-#include <cudf/strings/detail/utilities.cuh>
-#include <cudf/utilities/default_stream.hpp>
 #include <cudf/utilities/error.hpp>
-#include <cudf/utilities/memory_resource.hpp>
 #include <cudf/utilities/span.hpp>
 
 #include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
 #include <rmm/exec_policy.hpp>
 
-#include <thrust/iterator/transform_iterator.h>
-#include <thrust/pair.h>
+#include <cuda/functional>
 
 namespace cudf {
 
+namespace strings::detail {
+
 namespace {
-struct string_view_to_pair {
-  string_view null_placeholder;
-  string_view_to_pair(string_view n) : null_placeholder(n) {}
-  __device__ thrust::pair<char const*, size_type> operator()(string_view const& i)
-  {
-    return (i.data() == null_placeholder.data())
-             ? thrust::pair<char const*, size_type>{nullptr, 0}
-             : thrust::pair<char const*, size_type>{i.data(), i.size_bytes()};
+
+using string_pair         = thrust::pair<char const*, size_type>;
+using column_string_pairs = cudf::device_span<string_pair const>;
```

Comment:
```suggestion
using column_string_pairs = cudf::device_span<string_index_pair const>;
```
Use the existing `string_index_pair` already defined here: https://github.com/rapidsai/cudf/pull/17035/files#diff-5296ea4a99656673f3a0f601c847ab784edc45f28d21db840cbe060201482817R51